### PR TITLE
Sync SLO alerts with upstream kubernetes-mixin

### PR DESCRIPTION
**Pull request by dgrisonnet**
Sync our downstream SLO alert with the ones from upstream https://github.com/kubernetes-monitoring/kubernetes-mixin to make sure we benefit from their improvements. For instance, this could have prevented a few bugs that were found in 4.10.

### CODE REVIEW

File: jsonnet/jsonnetfile.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "version": 1,
+  "dependencies": [
+    {
+      "source": {
+        "git": {
+          "remote": "https://github.com/kubernetes-monitoring/kubernetes-mixin.git",
+          "subdir": ""
+        }
+      },
+      "version": "master"
```
Comment (dgrisonnet) [inline jsonnet/jsonnetfile.json:11]: Ideally, this should be pinned to release branches before OCP code freeze. https://github.com/kubernetes-monitoring/kubernetes-mixin#releases. release-0.10 is tied to Kubernetes 1.23 and OCP 4.10

File: Makefile [hunk 2]
```diff
@@ -100,3 +100,9 @@ verify-bindata-v4.1.0: verify-apirequestcounts-crd
 .PHONY: verify-apirequestcounts-crd
 verify-apirequestcounts-crd:
 	diff -Naup $(APIREQUESTCOUNT_CRD_SOURCE) $(APIREQUESTCOUNT_CRD_TARGET)
+
+ALERT_MANIFESTS := bindata/assets/alerts
+.PHONY: generate
+generate:
```
Comment (dgrisonnet) [inline Makefile:106]: I will add a step in CI to verify that the manifests are always up to date

File: Makefile [hunk 3]
```diff
@@ -100,3 +100,9 @@ verify-bindata-v4.1.0: verify-apirequestcounts-crd
...
+	jsonnet -J jsonnet/vendor -m $(ALERT_MANIFESTS) jsonnet/main.jsonnet | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml' -- {}
```
Comment (sanchezl) [inline Makefile:107]:
```suggestion
JSONNET:=go run github.com/google/go-jsonnet/cmd/jsonnet@v0.18.0
GOJSONTOYAML:=go run github.com/brancz/gojsontoyaml@v0.1.0
ALERT_MANIFESTS := bindata/assets/alerts
.PHONY: generate
generate:
	$(JSONNET) -J jsonnet/vendor -m $(ALERT_MANIFESTS) jsonnet/main.jsonnet | xargs -I{} sh -c 'cat {} | $(GOJSONTOYAML) > {}.yaml' -- {}
	find $(ALERT_MANIFESTS) -type f ! -name '*.yaml' -delete
```

What do you think if we do something like this to bring in dependent binaries?

**Review by sanchezl — COMMENTED** [review]
We split the slo alerts into basic and extended. The extended slos are not deployed on SNO due to the load it places on the single node.

File: bindata/assets/alerts/kube-apiserver-slos.yaml [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        long: 3d
+        severity: warning
+        short: 6h
+  - name: kube-apiserver-burnrate.rules
+    rules:
+    - expr: |
+        (
+          (
+            # too slow
+            sum by (cluster) (rate(apiserver_request_slo_duration_seconds_count{job="kube-apiserver",verb=~"LIST|GET",subresource!~"proxy|attach|log|exec|portforward"}[1d]))
```
Comment (sanchezl) [inline bindata/assets/alerts/kube-apiserver-slos.yaml:68]: We seem to use `job="apiserver"` in our copies.

File: bindata/assets/alerts/kube-apiserver-slos.yaml [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+    - alert: KubeAPIErrorBudgetBurn
+      annotations:
+        description: The API server is burning too much error budget.
+        summary: The API server is burning too much error budget.
+      expr: |
+        sum(apiserver_request:burnrate3d) > (1.00 * 0.01000)
+        and
+        sum(apiserver_request:burnrate6h) > (1.00 * 0.01000)
+      for: 3h
+      labels:
```
Comment (sanchezl) [inline bindata/assets/alerts/kube-apiserver-slos.yaml:58]: we add a `namespace: openshift-kube-apiserver` label.

File: bindata/assets/alerts/kube-apiserver-slos.yaml [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+        and
+        sum(apiserver_request:burnrate2h) > (3.00 * 0.01000)
+      for: 1h
+      labels:
+        long: 1d
+        severity: warning
+        short: 2h
+    - alert: KubeAPIErrorBudgetBurn
+      annotations:
+        description: The API server is burning too much error budget.
```
Comment (sanchezl) [inline bindata/assets/alerts/kube-apiserver-slos.yaml:51]: We extend the description and add `runbookUrl`:
```yaml
description: The API server is burning too much error budget. This alert fires when too many requests are failing with high latency. Use the 'API Performance' monitoring dashboards to narrow down the request states and latency. The 'etcd' monitoring dashboards also provides metrics to help determine etcd stability and performance.
runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/cluster-kube-apiserver-operator/KubeAPIErrorBudgetBurn.md
```

**Review by sanchezl — COMMENTED** [review]
Some more comments on the customization we apply to our copies.

**Comment by sanchezl** [comment]
The [runbook](https://github.com/openshift/runbooks/blob/master/alerts/cluster-kube-apiserver-operator/KubeAPIErrorBudgetBurn.md)  for the `KubeAPIErrorBudgetBurn`  might need to be updated if we are now in sync with upstream.

**Comment by dgrisonnet** [comment]
Closing, I will open another PR to introduce that change again.

**Pull request closed without merging**